boards/arm/circuitdojo_feather_nrf9160/Kconfig.defconfig

```diff
@@ -0,0 +1,39 @@
+# Circuit Dojo nRF9160 Feather configuration
+
+# Copyright (c) 2018-2020 Nordic Semiconductor ASA
+# Copyright (c) 2020 Circuit Dojo LLC
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_CIRCUITDOJO_FEATHER_NRF9160 || BOARD_CIRCUITDOJO_FEATHER_NRF9160_NS
+
+config BOARD
+	default "circuitdojo_feather_nrf9160"
+
+# For the secure version of the board the firmware is linked at the beginning
+# of the flash, or into the code-partition defined in DT if it is intended to
+# be loaded by MCUboot. If the secure firmware is to be combined with a non-
+# secure image (TRUSTED_EXECUTION_SECURE=y), the secure FW image shall always
+# be restricted to the size of its code partition.
+# For the non-secure version of the board, the firmware
+# must be linked into the code-partition (non-secure) defined in DT, regardless.
+# Apply this configuration below by setting the Kconfig symbols used by
+# the linker according to the information extracted from DT partitions.
+
+# Workaround for not being able to have commas in macro arguments
+DT_CHOSEN_Z_CODE_PARTITION := zephyr,code-partition
+
+config FLASH_LOAD_SIZE
+	default $(dt_chosen_reg_size_hex,$(DT_CHOSEN_Z_CODE_PARTITION))
+	depends on BOARD_CIRCUITDOJO_FEATHER_NRF9160 && TRUSTED_EXECUTION_SECURE
+
+if BOARD_CIRCUITDOJO_FEATHER_NRF9160_NS
+
+config FLASH_LOAD_OFFSET
+	default $(dt_chosen_reg_addr_hex,$(DT_CHOSEN_Z_CODE_PARTITION))
+
+config FLASH_LOAD_SIZE
+	default $(dt_chosen_reg_size_hex,$(DT_CHOSEN_Z_CODE_PARTITION))
```

Comment:
It looks like FLASH_LOAD_SIZE is always getting set to the size of `zephyr,code-partition`, regardless of whether it's a secure or non-secure configuration. The same default is set above.

Is there some potential cleanup to do here? If so I guess it should be done in a follow-up PR as this seems to be copy/pasted from the nRF9160DK Kconfig.defconfig. CC @anangl

Reply:
Hey @mbolivar-nordic,

You guys would know better than me. 😇 Don't the code partitions get resized based on the SPM in NCS?

Thanks for taking a look at this.

Reply:
> You guys would know better than me. innocent Don't the code partitions get resized based on the SPM in NCS?

Yep, it's a question for @anangl

Reply:
It looks fine. See https://github.com/zephyrproject-rtos/zephyr/commit/0122bacd2cc4b01909f991667cb17d67e92d2364. `FLASH_LOAD_SIZE` won't get this default when the image is built without `TRUSTED_EXECUTION_SECURE` being set.

Reply:
@anangl is right, @mbolivar-nordic  :)

Reply:
Sorry I was unclear. What I mean is something like this:

```
config FLASH_LOAD_OFFSET
	default $(dt_chosen_reg_addr_hex,$(DT_CHOSEN_Z_CODE_PARTITION)) if \
		BOARD_CIRCUITDOJO_FEATHER_NRF9160_NS

config FLASH_LOAD_SIZE
	default $(dt_chosen_reg_size_hex,$(DT_CHOSEN_Z_CODE_PARTITION)) if \
		BOARD_CIRCUITDOJO_FEATHER_NRF9160_NS ||
		(BOARD_CIRCUITDOJO_FEATHER_NRF9160 && TRUSTED_EXECUTION_SECURE)
```

Reply:
@mbolivar-nordic i fI read this correctly this is a "no-op" and looks correct to me, but i think it is a matter of taste at the end of the date. What worries me is that you might need to have the default setting in one loooong line.